pandas/tests/test_strings.py

```diff
@@ -367,11 +367,36 @@ def test_str_cat_align_mixed_inputs(self, join):
         with tm.assert_raises_regex(ValueError, rgx):
             s.str.cat([t, z], join=join)
 
-    def test_str_cat_raises(self):
```

Comment:
need a whole lot more testing here. bytes, all other index types (its possible some of these are tested abvove, if so, pls consolidate the tests).